src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
+                            !namespaceQualifiedName.StartsWith("Microsoft.WindowsAzure.Storage.", StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+
+                        var valueContentAnalysisResultOpt = ValueContentAnalysis.GetOrComputeResult(
+                                                                                    invocationOperation.GetTopmostParentBlock().GetEnclosingControlFlowGraph(),
+                                                                                    owningSymbol,
+                                                                                    wellKnownTypeProvider,
+                                                                                    interproceduralAnalysisConfig,
+                                                                                    out var copyAnalysisResult,
+                                                                                    out var pointsToAnalysisResult);
+                        var typeSymbol = methodSymbol.ContainingType;
+
+                        if (typeSymbol.Name != "CloudStorageAccount")
+                        {
+                            var protocolsArgumentOperation = invocationOperation.Arguments.FirstOrDefault(s => s.Parameter.Name == "protocols");
+
+                            if (protocolsArgumentOperation != null)
+                            {
+                                var protocolsArgument = valueContentAnalysisResultOpt[protocolsArgumentOperation.Kind, protocolsArgumentOperation.Syntax];
+
+                                if (protocolsArgument.IsLiteralState &&
+                                    !protocolsArgument.LiteralValues.Contains(1))
```

Comment:
>1 [](start = 78, length = 1)

Nit: Define a const for this, or at least provide a comment for what 1 is.

Reply:
Also please add a comment about how the constant `1` maps to your analysis/diagnostic message (`Use SharedAccessProtocol.HttpsOnly when creating Shared Access Signature (SAS)`). I am presuming it is an enum constant